### ISSUE TITLE
multilingual categories

### ISSUE THREAD

**Issue by andreav** [nitely/Spirit#145]
Hello.
Is it possible to translate categories?

Thank you.

**Comment by nitely** [comment]
you mean a category you have created? nop. Django does not support translating model fields out the box. [There are some package for this out there](https://github.com/django-parler/django-parler), but I'm hesitant to add another third-party app.

**Comment by andreav** [comment]
Yes, exactly, I mean categories i created.

Don't you think multilingual category names are essential for spirit
translation?

By the way, also https://github.com/deschler/django-modeltranslation works
great.






Il 27/dic/2016 23:14, "Esteban Castro Borsani" <notifications@github.com>
ha scritto:

**Comment by nitely** [comment]
> Don't you think multilingual category names are essential for spirit translation?

No. Mostly coz the threads will be in a given language. Say your category names are translated to Italian, what about the threads/comments? threads may still be in english or whatever. I actually don't know of a forum that supports translating category names (there probably are, though).

> https://github.com/deschler/django-modeltranslation 

Saw that. But this is probably one of the few places where I'd not normalize, otherwise you end up with many extra empty fields (ie: 10 empty fields or more), I dunno if that is such a good idea. Normalizing + caching seems better.

**Comment by andreav** [comment]
I don't know, I was thinking to this scenario: a user starts an Italian
topic. He writes Italian and would prefer choosing among Italian categories.
In this scenario,  who starts the topic also sets its language.
But maybe this is not the case.






Il 28/dic/2016 00:11, "Esteban Castro Borsani" <notifications@github.com>
ha scritto:

Don't you think multilingual category names are essential for spirit
translation?

No. Mostly coz the threads will be in a given language. Say your category
names are translated to Italian, what about the threads/comments? threads
may still be in english or whatever. I actually don't know of a forum that
supports translating category names (there probably are, though).

https://github.com/deschler/django-modeltranslation

Saw that. But this is probably one of the few places where I'd not
normalize, otherwise you end up with many extra empty fields (ie: 10 empty
fields or more), I dunno if that is such a good idea. Normalizing + caching
seems better.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/nitely/Spirit/issues/145#issuecomment-269396080>, or mute
the thread

**Comment by nitely** [comment]
> He writes Italian and would prefer choosing among Italian categories.

But then all category names would be translated to italian, so which one it's actually for italian topics?
I think in that use case, it works fine as it's now. You create categories with english names for topics in english and categories with italian names for topics in italian.

Also, if category names are translated to *my* language, spanish for example, I would expect the topics to be in spanish, not in other language.

**Comment by andreav** [comment]
Sorry if I continue, but believe me,  I'm trying to truly understand your point of view in order to best leverage spirit features.

> But then all category names would be translated to italian, so which one it's actually for italian topics?

Yes, categories in my opinion would be cross language. Any category could be used for a topic, then the topic will sometimes be written in Italian, sometimes be written in English (according to who creates it)

> You create categories with English names for topics in English and categories with Italian names for topics in Italian

In spirit, how could I differentiate between English or Italian topics? I was thinking this separation was just because I publish a topic written in Italian and someone else publishes a topic written in English.
Are you suggesting to create the same category many times but in different languages? But so, when I create a topic, I will have to choose (from the category drop-down) among multiple translated versions of the same category.

> if category names are translated to my language, spanish for example, I would expect the topics to be in spanish, not in other language.

Yes, but this is another problem: How can I translate topics content in the language of who is viewing the page (in order to let anyone read this topic)?

Thank you again.

**Comment by andreav** [comment]
Ok, now I see some pros and cons I did not realise before.
Thank you!

**Comment by nitely** [comment]
Let me know how you end up solving this. Feel free to re-open :smiley_cat:

**Comment by andreav** [comment]
Analysing my site use case, users look for topics primary by geographic position.
I hope/suppose they are more interested in finding someone speaking about a particular place, rather than in their language. Then browsers (or some javascript I could put on the page) will approximately translate the rest of the topic content for them.

In this scenario, I would like to share the same category among languages. This will create the mixed language topic list you spoke about before, but this is quite desired (because the list will be filtered first by position)

I just implemented multilingual spirit.category.title using [django-modeltranslation](https://github.com/deschler/django-modeltranslation) (this is the one I know, but I think it makes a very good job).

I created:

- translation.py file (next to spirit.category.models.py) specifying category title field is multilingual
- admin.py registering spirit category at admin interface. This is done to have a place where to input title translations  (spirit admin category interface does not show title translation alternatives)

It works like a charm!
At least, I'm curios to see how this forum will evolve!
